CONTRIBUTING.md

```diff
@@ -1,51 +1,147 @@
-## Local development
+## Table of Contents
 
-### Pre-requisites
+- [Setting Up Local Development Environment](#setting-up-local-development-environment)
+  - [Prerequisites for Development](#prerequisites-for-development)
+  - [Building, Running, and Testing the Repository](#building-running-and-testing-the-repository)
+  - [Setting Up Local Language Servers](#setting-up-local-language-servers)
+    - [Roslyn](#roslyn)
+    - [Razor](#razor)
+  - [Configuring Local Language Servers](#configuring-local-language-servers)
+    - [Roslyn](#roslyn-1)
+    - [Razor](#razor-1)
+  - [Debugging Local Language Servers](#debugging-local-language-servers)
+- [Creating VSIX Packages for the Extension](#creating-vsix-packages-for-the-extension)
+- [Updating the `Roslyn` Language Server Version](#updating-the-roslyn-language-server-version)
 
-First install:
+## Setting Up Local Development Environment
+
+Setting up your local development environment for the vscode-csharp repository involves several steps. This guide will walk you through the process.
+
+### Prerequisites for Development
+
+Before you start, make sure you have the following software installed on your machine:
 
 * Node.js ([v18.17.0 LTS](https://nodejs.org/en/blog/release/v18.17.0) is recommended).
 * Npm (The version shipped with node is fine)
-* .NET 7.0 SDK (dotnet should be on your path)
+* .NET 8.0 SDK (dotnet should be on your path)
```

Comment:
Technically .net8 should work, but IMHO we should stay with recommending .net7 for now since
1.  CI uses .net 7
2.  the extension uses a .net7 runtime still.

Reply:
```suggestion
* .NET 7.0 SDK (dotnet should be on your path)
```